### ISSUE TITLE
File extension '' not allowed

### ISSUE THREAD

**Issue by matter3** [klis87/django-cloudinary-storage#5]
I'm having some issues using ImageField in models. After uploading an image via an admin form and then going back to do some edits on that newly created entity, I get `File extension '' is not allowed...` when hitting Save. 

Is there a good way around this? Right now I feel like anything I try to do to avoid this error is hackish.

Thanks for the lib by the way, really helping me out!

**Comment by klis87** [comment]
Thanks for raising this issue, I didnt catch this problem myself. And I am glad you find this library useful :)

Regarding your issue, the problem is that we shouldn't save file extension in a file name, as you would end up with duplicated extension in Cloudinary url, like file.jpg.jpg. This is because Cloudinary allows you to convert images to different extensions on the fly, and as I remember, in case of file.jpg as Cloudinary file public ID, generated url would be file.jpg.jpg , as jpg is the default for Cloudinary.

As a workaround, I guess the best would be to add a custom validator to image fields, which allows ```''``` extension. It would be beneficial to add this to the package README as well. What do you think about this solution and what did you try yourself?

**Comment by klis87** [comment]
Actually I tried this right now I cannot recreate this issue. I have a following model:
```
class TestImageModel(models.Model):
    name = models.CharField(max_length=100)
    file = models.FileField(upload_to='tests/', blank=True, storage=RawMediaCloudinaryStorage())
    image = models.ImageField(upload_to='tests-images/', storage=MediaCloudinaryStorage())
```

When I save a new instance via the admin panel with an uploaded image, I can always go back to it and edit different fields like name without any error. I can switch image to a different one as well without any problem.

If you still have below problem, please comment on this issue. I am closing for now.

**Comment by matter3** [comment]
Following up on this since I have returned from vacation. It appears this was an issue that has been fixed in Django 1.11.2. See release notes, last bullet: https://docs.djangoproject.com/en/1.11/releases/1.11.2/

Everything works now, thanks for the help.
